### ISSUE TITLE
It's not perhaps a bug, but unwanted behavior (face, vert)

### ISSUE THREAD

**Issue by GiovanniSer** [cnr-isti-vclab/meshlab#967]
Win 10, running ML ver. 2020.12 and only the previous one.
Well, we assume that I have several faces rendered in "face" mode (flat).
Then I call the "Re-Orient all faces coherentely".
The command works fine, but the rendered mode changes in "vert" (nuanced).
It happens also with other commands ( i.e. "Invert faces orientations").
In my opinion (I load often and often I do these operations) it is unnerving and not justified.
Visual analysis of geometry is impaired, it is not expected and so it requires a subsequent view adjustment command.
Thanks in advanced ...
G.

**Comment by alemuntoni** [comment]
Confirmed on every platform. I agree, it is unexpected and it should be fixed. 
Hope to find time to investigate this soon! :)

**Comment by momentmal** [comment]
I can report the same problem - I have a PLY with vertex colors and when trying to clean up the mesh with Fill Color set to "Mesh" (basically a grey clay type of view mode) instead of "Vertex", it always switches back to "vertex" Fill rendering mode, which is really not what I want to see when cleaning up vertices in a Scan.

**Comment by momentmal** [comment]
Ok this happens on pretty much any operation in Meshlab. Especially annoying if I have mulitple layers, which I all changed to Mesh coloring - they ALL switch back to displaying default Vertex Colors after any operation...

**Comment by GiovanniSer** [comment]
You are right, other edit operations cause the face normals to be swapped in the direction of the vertices. I can't be precise because I use them few times.
Sometimes the problem involves only the render window. The icons remain to indicate the previous mode and therefore they appears misaligned with the presentation mode of the object.
Thanks.

**Comment by momentmal** [comment]
Honestly I don't know what you are trying to say, I was adressing the changing of the rendered mode after doing any operation. The settings set on each layer should just stay where they are instead of resetting themselves after any operation
![image](https://user-images.githubusercontent.com/9054832/118361399-9633f080-b58b-11eb-8e7b-f6c19fd6fced.png)

**Comment by momentmal** [comment]
@alemuntoni are you aware of an old build which doesn't have this unwanted behavior? it's really driving me insane when trying to clean up a scan 😅 any way I can download a specific old build?

**Comment by GiovanniSer** [comment]
I was talking about these icons
![Clipboard01](https://user-images.githubusercontent.com/67306481/118361771-2161b600-b58d-11eb-97f1-d7dc1b714304.jpg)

**Comment by alemuntoni** [comment]
> @alemuntoni are you aware of an old build which doesn't have this unwanted behavior? it's really driving me insane when trying to clean up a scan  any way I can download a specific old build?

The behaviour you described is there since 2016.12, therefore there are no builds without it..

**Comment by alemuntoni** [comment]
The issue described by @GiovanniSer has been solved in the last commit.
The issue described by @momentmal is still there, it requires further investigation.

**Comment by momentmal** [comment]
> The issue described by @GiovanniSer has been solved in the last commit.
> The issue described by @momentmal is still there, it requires further investigation.

Thanks for the update! And sorry for attaching slightly different issue to this ticket, should i create a new ticket for this? Really looking forward to a fix for this.

**Comment by alemuntoni** [comment]
Don't worry, it's fine to keep this ticket! Anyway, I am confident that it will be fixed soon.

**Comment by alemuntoni** [comment]
Finally fixed :)

**Comment by momentmal** [comment]
> Finally fixed :)

fantastic news! when is a new release build containing this fix expected to be available for download? 
I currently don't have a building environment setup here.

**Comment by alemuntoni** [comment]
I'll try to release 2021.05 today or next week!

**Comment by momentmal** [comment]
Thanks for the quick new release @alemuntoni!

- Shading settings of other layers now stays consistent on operations like "merge close vertices".
- However, on duplicating a layer, the shading of the existing layer as well as the duplicated layer changes. This behavior is unwanted. Example in screenshots below.

![image](https://user-images.githubusercontent.com/9054832/119174889-7019bf00-ba69-11eb-9679-ff1424b1348f.png)
1.  Source PLY layer with these custom shading settings

![image](https://user-images.githubusercontent.com/9054832/119174967-89227000-ba69-11eb-98fb-5ab5045f2917.png)
2. After duplicating, the source layer changes its shading settings to this

![image](https://user-images.githubusercontent.com/9054832/119175011-98092280-ba69-11eb-875b-aaabe8ded8f2.png)
3. The duplicated layer has these different shading settings

**Comment by momentmal** [comment]
@alemuntoni just wanted to make sure you saw my comments on testing the shading setting behavior on your new release build?

**Comment by alemuntoni** [comment]
Sorry for the late reply. Yes, I saw that. I'll try to fix also this soon

**Comment by momentmal** [comment]
> Could you please try the last nightly build?

How do I download it? I have no idea how to find it.

**Comment by alemuntoni** [comment]
Sorry, in the actions tab! [here](https://github.com/cnr-isti-vclab/meshlab/wiki/How-to-intall-the-last-nightly) there are the instructions

**Comment by momentmal** [comment]
Ok so it improved a bit: The layer i duplicate now keeps the set shading properties.

![image](https://user-images.githubusercontent.com/9054832/120002069-baa0bb80-bfd4-11eb-8ad3-b4f93ab5cdde.png)

But the duplicated layer doesn't keep the same shading properties, it kinda seems to go back to Meshlabs default setting or something:

![image](https://user-images.githubusercontent.com/9054832/120002713-63e7b180-bfd5-11eb-85a9-fad1b6f93f70.png)

If you e.g. incrementally reduce and clean your mesh you want to keep certain stages as layers, but there is no reason that duplicates of you layer should change its shading bevavior without specific user input.

**Comment by alemuntoni** [comment]
I agree with you, but on this I can't do much. 

By design, the filter that duplicates the layer does not manage shading options (they are managed by the meshlab framework), but at the same time the meshlab framework does not know that the newly created mesh is a duplicate of another one.

To solve this, I should make some kind of revolution in the codebase of meshlab :)

**Comment by momentmal** [comment]
Done. 

Btw, hey, i opened issue #1000 🥳🥳 🙃

**Comment by alemuntoni** [comment]
> Btw, hey, i opened issue #1000
 
saw that :D I don't know if it is a good or a bad thing though :D
In any case, we are (slowly) improving meshlab!
Thanks for the help, @momentmal :)
